keps/sig-cloud-provider/2699-add-webhook-hosting-to-ccm/README.md

```diff
@@ -0,0 +1,541 @@
+specific webhooks. The intent is to make it easy to either generate a binary
+or enhance the existing CCM binary to host such webhooks. We also intend to
+allow for easily linking in "standard" webhooks needed by other SIGs which
+need to be customized for particular cloud providers.
+
+## Motivation
+
+The Cloud (Provider) Controller Manager (CCM) is the binary into which the Cloud
+Provider places all the controllers needed to make a Kubernetes cluster work
+correctly on their Cloud. There are also occasions when it makes sense for a Cloud
+Provider to want these customizations to be applied inline rather than 
+asynchronously after a change has already been applied. 
+
+Our initial example of this is from SIG Storage. These would like the functionality 
+from PVL admission controller (https://github.com/kubernetes/kubernetes/issues/52617).
+This needs to be completed for cloud provider extraction to complete. Several
+Cloud Providers have indicated that this should be done inline, especially as the
+existing deprecated solution is an inline solution.
+
+### Goals
+
+Our imediate goal is to allow in tree Cloud Providers are able to stop using the 
+existing PVL admission controller and do so using the framework. However we want to
+build a framework which wil be usable by similar solutions to problems. This KEP is
+about the framework needed to support the PVL webhook and not the webhook itself.
+
+### Non-Goals
+
+We are not intending to create a [general admission webhook solution](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html). This is just
+intended to host Cloud Provider specific webhooks as part of the Control Plane.
+
+## Proposal
+
+We will start by adding extension hooks which can be registered in the 
+cmd/cloud-controller-manager/main.go. This would be similar to the mechanism
+we already use to register new controllers. The existing sample shows this with
+a sample of registering the nodeipamcontroller which is not a normally installed
+controller in the cloud controller manager. In a similar way we will have a 
+sample of integrating a PVL mutating webhook into the sample CCM. We will also 
+have the system automatically detect if there are both controllers and webhooks
+registered in the binary. If both are registered it will automatically add 
+command line flags allowing webhooks and controller to be disabled. The 
+controller flag will default to the controllers being enabled. The webhook flag
+will default to the webhooks being disabled. We would also like to provide a 
+builder pattern for registering both the controller and webhook extensions.
+
+Another issue to consider is how the mutating/asmission webhook configuration is
+written into the cluster. This may be somewhat dependent on if the Cloud Provider
+intends to run it on the Control Plane or on the Cluster. We would recommend on 
+running in on the Control Plane. However for some Coud Providers that can lead to
```

Comment:
This paragraph is becoming hard to follow because we keep referring to the subject as "it". I find in enhancement docs it's better to be explicit, eg `This may be somewhat dependent on if the Cloud Provider
intends to run the webhook on the Control Plane or on the Cluster.`

Aside from that, and maybe this is a lack of context on my part, I'm not sure why there's a distinction here, are you talking about whether they run the webhook on control plane nodes vs worker nodes? Why would that make a difference? Can we expand on that within the Kep somewhere?

Reply:
I'm usually the one complaining about the problem of using 'it' :smiley_cat: Fixed. 

Added another paragraph with at least some of the concerns specific to running the webhook server in the cluster (in the nodes). I realize there is a little blurring of control plane vs worker and static vs dynamic. I can change if you think it needs to be called out. However at this point I don't want to over complicate this section.